clarity/contracts/pool-token/ytp-yield-wbtc-59760-wbtc.clar

```diff
@@ -1,89 +1,78 @@
-(impl-trait .trait-sip-010.sip-010-trait)
 (impl-trait .trait-pool-token.pool-token-trait)
 
-(define-fungible-token yield-wbtc-59760-wbtc)
+(define-fungible-token ytp-yield-wbtc-59760-wbtc)
 
 (define-data-var token-uri (string-utf8 256) u"")
-(define-data-var contract-owner principal tx-sender)
+(define-data-var contract-owner principal .yield-token-pool)
 
 ;; errors
 (define-constant ERR-NOT-AUTHORIZED (err u1000))
 
-
-(define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
-  (match (ft-transfer? yield-wbtc-59760-wbtc amount sender recipient)
-    response (begin
-      (print memo)
-      (ok response)
-    )
-    error (err error)
+(define-public (set-contract-owner (owner principal))
+  (begin
+    (asserts! (is-eq tx-sender (var-get contract-owner)) ERR-NOT-AUTHORIZED)
+    (ok (var-set contract-owner owner))
```

Comment:
Should we settle on explicit conformity with the `ownable-trait` (in STXDX)?

Reply:
Makes sense.